app/views/course/assessment/question/programming/_form.html.slim

```diff
@@ -1,24 +1,2 @@
-= simple_form_for [current_course, @assessment, @programming_question] do |f|
-  = f.error_notification
-  = import_result_alert
-  = render partial: 'course/assessment/questions/form', locals: { f: f }
-
-  = f.association :language
-  = f.input :memory_limit
-  = f.input :time_limit
-  - unless @assessment.autograded?
-    = f.input :attempt_limit, placeholder: t('.attempt_limit_placeholder')
-  = f.label :template_package
-  = f.attachment
-
-  - if display_build_log?
-    = render partial: 'form_build_log',
-             locals: { log: @programming_question.import_job.error.slice('stdout', 'stderr') }
-
-  = render partial: 'form_templates',
-           locals: { programming_question: @programming_question }
-
-  = render partial: 'form_test_cases',
-           locals: { test_cases: @programming_question.test_cases }
-
-  = f.button :submit
+div#programming-question
+  = fa_icon 'spinner 4x spin'
```

Comment:
Don't use an extra request to load the form. You can still render the JSON props and make it the attribute of a div, and read the JSON from div in the frontend. 

Or you could find better ways to do that.